src/Veldrid.Tests/RenderTests.cs

```diff
@@ -860,6 +858,70 @@ public void ComputeGeneratedTexture()
             GD.Unmap(readback);
         }
 
+        [SkippableTheory]
+        [InlineData(2)]
+        [InlineData(6)]
+        public void ComputeBindTextureWithArrayLayersAsWriteable(uint ArrayLayers)
+        {
+            Skip.IfNot(GD.Features.ComputeShader);
+
+            uint TexSize = 32;
+            uint MipLevels = 1;
+            TextureDescription texDesc = TextureDescription.Texture2D(
+                TexSize, TexSize,
+                MipLevels,
+                ArrayLayers,
+                PixelFormat.R8_G8_B8_A8_UNorm,
+                TextureUsage.Sampled | TextureUsage.Storage);
+            Texture computeOutput = RF.CreateTexture(texDesc);
+            
+            ResourceLayout computeLayout = RF.CreateResourceLayout(new ResourceLayoutDescription(
+                new ResourceLayoutElementDescription("ComputeOutput", ResourceKind.TextureReadWrite, ShaderStages.Compute)));
+            ResourceSet computeSet = RF.CreateResourceSet(new ResourceSetDescription(computeLayout, computeOutput));
+
+            Pipeline computePipeline = RF.CreateComputePipeline(new ComputePipelineDescription(
+                TestShaders.LoadCompute(RF, "ComputeImage2DArrayGenerator"),
+                computeLayout,
+                32, 32, 1));
+
+            CommandList cl = RF.CreateCommandList();
+            cl.Begin();
+            cl.SetPipeline(computePipeline);
+            cl.SetComputeResourceSet(0, computeSet);
+            cl.Dispatch(TexSize / 32, TexSize / 32, ArrayLayers);
+            cl.End();
+            GD.SubmitCommands(cl);
+            GD.WaitForIdle();
+
+            float sideColorStep = 1.0f / ArrayLayers;
+            Texture readback = GetReadback(computeOutput);
+            Assert.All
+            (
+                from mip in Enumerable.Range(0, (int)MipLevels)
+                from layer in Enumerable.Range(0, (int)ArrayLayers)
+                select (mip, layer),
+                (t) =>
+                {
+                    var subresource = readback.CalculateSubresource((uint)t.mip, (uint)t.layer);
+                    var mipSize = TexSize >> t.mip;
+                    var expectedColor = (byte)255.0f * ((t.layer + 1) * sideColorStep);
+                    var map = GD.Map<byte>(readback, MapMode.Read, subresource);
+
+                    Assert.All(
+                        from x in Enumerable.Range(0, (int)mipSize)
+                        from y in Enumerable.Range(0, (int)mipSize)
+                        select (X: x, Y: y),
+                        (xy) =>
+                        {
+                            Assert.InRange(map[xy.X, xy.Y], expectedColor - 1, expectedColor + 1);
```

Comment:
This is dumb.  It's working around undefined rounding rules coming back from the math in the shader.

I can probably pass values into the shader instead.  If you've got a better suggestion, let me know.

Reply:
`floor` worked of course, thanks! :+1: